org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/MovingAverage.java

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal;
+
+public class MovingAverage {
```

Comment:
I would probably add a Javadoc here to explain that this is a moving average where the first iteration becomes the first term passed in. That way when someone sees the  `this.n` on the denominator, they don't assume it's wrong, based on https://en.wikipedia.org/wiki/Moving_average#Cumulative_moving_average .

Reply:
Good idea.